Squared/Threading/Future.cs

```diff
@@ -190,6 +192,8 @@ public SignalFuture (bool signaled)
 
     public static class Future {
         public enum State : int {
+            // REVIEW: I guess Unknown and Disposing are removed because
+            // you did not want these states to be part of the public API?
```

Comment:
Yeah I don't want the consumer to look at those and try to do something based on seeing them, though it might make sense to just broadcast them as 'Indeterminate' or something and say you should treat those as a 'please retry later'

Reply:
I think that, as both `State_Indeterminate` and `State_Disposing` are transitory, not exposing them is what makes the most sense for everyone (in particular, `State_Indeterminate` should really be understood as "This `Future` instance is locked with regard to operations that could resolve it", but no client that is outside of `Future` should treat it differently from "not resolved yet", which is `Empty`).

However, I guess that, given the existence of the various boolean properties (`Resolved`/`Completed`/`Disposed`/`CompletedSuccessfully`/`CompletedWithError`), I am not 100% sure that this `enum` needs to be public?

Reply:
(Of course, I am not saying that it should now be removed - that would break existing clients for no good reason)